app/queries/get_unsubmitted_applications_ready_to_nudge.rb

```diff
@@ -1,43 +1,37 @@
 class GetUnsubmittedApplicationsReadyToNudge
-  MAILER = 'candidate_mailer'.freeze
-  MAIL_TEMPLATE = 'nudge_unsubmitted'.freeze
+  # The purpose of this nudge is to contact candidates who have
+  # - completed their application forms,
+  # - made an application choice
+  # - but NOT submitted anything.
   COMMON_COMPLETION_ATTRS = (ApplicationForm::SECTION_COMPLETED_FIELDS - %w[science_gcse efl course_choices])
     .map { |field| "#{field}_completed" }.freeze
 
   def call
     uk_and_irish_names = NATIONALITIES.select do |code, _name|
       code.in?(ApplicationForm::BRITISH_OR_IRISH_NATIONALITIES)
     end.map(&:second)
-    uk_and_irish = uk_and_irish_names.map { |name| ActiveRecord::Base.connection.quote(name) }.join(',')
 
     ApplicationForm
+      # Only candidates with unsubmitted application_choices
+      .joins(:application_choices).where('application_choices.status': 'unsubmitted')
+      # Filter out candidates who should not receive emails about their accounts
+      .joins(:candidate).where(candidates: { submission_blocked: false, account_locked: false, unsubscribed_from_emails: false })
+      # Filter out anyone who has already received this message
+      .joins("LEFT OUTER JOIN emails ON emails.application_form_id = application_forms.id AND emails.mailer = 'candidate_mailer' AND emails.mail_template = 'nudge_unsubmitted'")
+      .where(emails: { id: nil })
```

Comment:
I think there are some performance issues with the scope (it is ok with the other ones because there aren't as many other joins). But I will have a look.

Reply:
I've replaced that.